test/unit/util/mapbox.test.js

```diff
@@ -322,82 +304,219 @@ test("mapbox", (t) => {
 
-        t.test('POSTs appuserTurnstile event when access token changes', (t) => {
-            config.ACCESS_TOKEN = 'pk.new.*';
+                t.stub(browser, 'now').callsFake(() => now + 5); // A bit later
 
-            mapbox.postTurnstileEvent(['a.tiles.mapbox.com']);
+                mapbox.postTurnstileEvent(['a.tiles.mapbox.com']);
 
-            const req = window.server.requests[0];
-            req.respond(200);
+                t.false(window.server.requests.length);
+                t.end();
+            });
 
-            const reqBody = JSON.parse(req.requestBody)[0];
-            t.equal(req.url, `${config.EVENTS_URL}?access_token=pk.new.*`);
-            t.equal(req.method, 'POST');
-            t.equal(reqBody.event, 'appUserTurnstile');
-            t.equal(reqBody.sdkVersion, version);
-            t.ok(reqBody.userId);
+            t.test('POSTs event when previously stored anonId is not a valid uuid', (t) => {
+                config.ACCESS_TOKEN = 'pk.*';
+                const now = +Date.now();
 
-            t.end();
-        });
+                window.localStorage.setItem(`mapbox.turnstileEventData:${config.ACCESS_TOKEN}`, JSON.stringify({
+                    anonId: 'anonymous',
+                    lastSuccess: now
+                }));
 
-        t.test('POSTs appuserTurnstile event on next calendar day', (t) => {
-            config.ACCESS_TOKEN = 'pk.*';
-            const today = Date.now();
-            const tomorrow = today + (25 * 60 * 60 * 1000); // Add a day
-            t.stub(browser, 'now').callsFake(() => tomorrow);
+                t.stub(browser, 'now').callsFake(() => now + ms25Hours); // next day
+
+                mapbox.postTurnstileEvent(['a.tiles.mapbox.com']);
+
+                const req = window.server.requests[0];
+                req.respond(200);
+
+                const reqBody = JSON.parse(req.requestBody)[0];
+                t.notEqual(reqBody.userId, 'anonymous');
+                t.end();
+            });
+
+            t.test('POSTs event when previously stored timestamp is in the future', (t) => {
+                config.ACCESS_TOKEN = 'pk.*';
+                const now = +Date.now();
+
+                window.localStorage.setItem(`mapbox.turnstileEventData:${config.ACCESS_TOKEN}`, JSON.stringify({
+                    anonId: uuid(),
+                    lastSuccess: now + ms25Hours // 24-hours later
+                }));
+
+                t.stub(browser, 'now').callsFake(() => now); // Past relative ot lastSuccess
+
+                mapbox.postTurnstileEvent(['a.tiles.mapbox.com']);
+
+                const req = window.server.requests[0];
+                req.respond(200);
+                const reqBody = JSON.parse(req.requestBody)[0];
+                t.ok(reqBody.created, new Date(now).toISOString());
+
+                t.end();
+            });
+
+            t.test('does not POST appuserTurnstile event second time within same calendar day', (t) => {
```

Comment:
Rather than having some tests rely on state from previous tests, I'd be in favor of making each test self-contained. 
 This could be done by moving the logic from `mapbox.postTurnstileEvent` into a method on `TurnstileEvent`, exporting the `TurnstileEvent` class, and having each test construct a new instance to test against.

Reply:
I understand that it will make the tests self contained, but I think it is better to have a singleton `TurnstileEvent` for each instance of the map so that there can be a local cache of the eventData. Exposing the entire class without `postTurnstileEvent` being static would require the callers (Map in this case) to manage an instance of the event which seems unnecessary

Reply:
Exposing the class doesn't preclude having a singleton:

```js
export class TurnstileEvent {
   ...
   postTurnstileEvent(...) { /* actual logic */ }
}

const _turnstileEvent = new TurnstileEvent();

export function postTurnstileEvent() {
  return _turnstileEvent.postTurnstileEvent(...);
}
```